jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/DelayedHandlerTest.java

```diff
@@ -337,6 +347,62 @@ public boolean handle(Request request, Response response, Callback callback) thr
         }
     }
 
+    @Test
+    public void testNoDelayWithChunkedContent() throws Exception
+    {
+        DelayedHandler delayedHandler = new DelayedHandler();
+
+        _server.setHandler(delayedHandler);
+        delayedHandler.setHandler(new HelloHandler()
+        {
+            @Override
+            public boolean handle(Request request, Response response, Callback callback) throws Exception
+            {
+                // Check that we are called directly from HttpConnection.onFillable
+                ByteArrayOutputStream out = new ByteArrayOutputStream(8192);
+                new Throwable().printStackTrace(new PrintStream(out));
+                String stack = out.toString(StandardCharsets.ISO_8859_1);
+                assertThat(stack, containsString("org.eclipse.jetty.server.internal.HttpConnection.onFillable"));
+                assertThat(stack, containsString("org.eclipse.jetty.server.handler.DelayedHandler.handle"));
```

Comment:
Those assertions look very brittle. Can't we assert that a single `Request.read()` gets all the content with only one call instead? And maybe also assert that it isn't the case when `DelayedHandler` isn't used?

Reply:
Oh it is!  Thinking about how else to do it....

Reply:
They're somewhat okay as you're checking for the existence of a method name in a stack trace, so if we ever renamed a method, this test would fail. This is less brittle than `testDelayedUntilContent` which checks that a method name is not present, so if we ever rename it that test will incorrectly pass.

Maybe file a new issue to clean this up, this isn't wort blocking this PR IMHO.